tests/utilities/graphqlConnectionFactory.spec.ts

```diff
@@ -1,12 +1,180 @@
-import { describe, expect, it } from "vitest";
-import { graphqlConnectionFactory } from "../../src/utilities/graphqlConnectionFactory";
+import { beforeAll, describe, expect, it } from "vitest";
+import {
+  graphqlConnectionFactory,
+  getLimit,
+  getFilterObject,
+  getSortingObject,
+  generateConnectionObject,
+} from "../../src/utilities/graphqlConnectionFactory";
+import { type CursorPaginationInput } from "../../src/types/generatedGraphQLTypes";
+import { Types } from "mongoose";
+import { nanoid } from "nanoid";
 
-describe("utilities -> graphqlConnectionFactory", () => {
+describe("utilities -> graphqlConnectionFactory -> graphqlConnectionFactory", () => {
   it(`Returns a connection object with default/pre-defined fields which
 represents a connection that has no data at all and cannot be paginated.`, () => {
     const connection = graphqlConnectionFactory<unknown>();
 
     expect(connection).toEqual({
+      edges: [],
+      pageInfo: {
+        startCursor: null,
+        endCursor: null,
+        hasNextPage: false,
+        hasPreviousPage: false,
+      },
+    });
+  });
+});
+
+describe("utilities -> graphqlConnectionFactory -> getLimit", () => {
+  it(`Should return 1 + limit if the cursor is not supplied`, () => {
+    const args: CursorPaginationInput = {
+      limit: 10,
+      direction: "FORWARD",
+    };
+
+    expect(getLimit(args)).toBe(11);
+  });
+
+  it(`Should return 2 + limit if the cursor is supplied`, () => {
+    const args: CursorPaginationInput = {
+      cursor: "123456",
+      limit: 10,
+      direction: "FORWARD",
+    };
+
+    expect(getLimit(args)).toBe(12);
+  });
+});
+
+describe("utilities -> graphqlConnectionFactory -> getSortingObject", () => {
+  it(`Should return the supplied sorting object when the direction is forward`, () => {
+    const sortingObject = {
+      a: 1,
+      b: 1,
+    };
+
+    const payload = getSortingObject("FORWARD", sortingObject);
+
+    expect(payload).toEqual({
+      a: 1,
+      b: 1,
+    });
+  });
+
+  it(`Should return the supplied sorting object negated and in the same order when the direction is backward`, () => {
```

Comment:
i'm skeeptical about the order thing

Reply:
I had tried using the same locally, and it seemed to work well for simple cases. For more complex cases with multiple fields, the implementation of the resolver would be custom, and thus this connection factor wouldn't be used. 

Is there anything you feel that needs to be added to the same?